optuna/integration/pytorch_distributed.py

```diff
@@ -0,0 +1,252 @@
+from datetime import datetime
+import pickle
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+
+import optuna
+from optuna._experimental import experimental
+from optuna._imports import try_import
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalChoiceType
+
+
+with try_import() as _imports:
+    import torch
+    import torch.distributed as dist
+
+
+@experimental("2.6.0")
+class TorchDistributedTrial(optuna.trial.BaseTrial):
+    """A wrapper of :class:`~optuna.trial.Trial` to incorporate Optuna with PyTorch distributed.
+
+    .. seealso::
+        :class:`~optuna.integration.TorchDistributedTrial` provides the same interface as
+        :class:`~optuna.trial.Trial`. Please refer to :class:`optuna.trial.Trial` for further
+        details.
+
+    See `the example <https://github.com/optuna/optuna/blob/master/
+    examples/pytorch_distributed_simple.py>`__
+    if you want to optimize an objective function that trains neural network
+    written with PyTorch distributed data parallel.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` object or :obj:`None`. Please set trial object in
+            rank-0 node and set :obj:`None` in the other rank node.
+        device:
+            A `torch.device` to communicate with the other nodes. Please set a CUDA device
+            assigned to the current node if you use "nccl" as `torch.distributed` backend.
+
+    .. note::
+        The methods of :class:`~optuna.integration.TorchDistributedTrial` are expected to be
+        called by all workers at once. They invoke synchronous data transmission to share
+        processing results and synchronize timinig.
```

Comment:
```suggestion
        processing results and synchronize timing.
```

Reply:
Oops! I applied the suggestion in https://github.com/optuna/optuna/pull/2303/commits/98ed9de6a948462fb65651ab43af1e57cc99706d. Thank you!